graphql-dgs/src/main/java/com/netflix/graphql/dgs/Trivial.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2024 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.graphql.dgs;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import graphql.TrivialDataFetcher;
+import graphql.schema.DataFetcher;
+
+/**
+ * Marker interface to signal that a {@link DgsData} annotated method is "trivial", that is
+ * that the underlying {@link DataFetcher} should implement {@link TrivialDataFetcher}. Useful
+ * to connote simple data fetcher methods that don't do any I/O for instance; such methods are
+ * often handled specially by instrumentation.
+ *
+ * @see TrivialDataFetcher
+ */
+@Target({ElementType.METHOD, ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Inherited
+public @interface Trivial { }
```

Comment:
`DgsTrivial`?

I'm torn on this, as I never liked the prefix, but it's on _nearly all_ of the interfaces. And with the spring-graphql integration, I feel like the inconsistency might add some initial readability confusion.

Reply:
Yeah I understand that concern, I ended up revisiting just adding a new field to the existing annotations based on Paul's feedback, and was able to make that work, so I'm scrapping this new annotation anyway.